src/main/java/org/opensearch/ad/task/ADTaskManager.java

```diff
@@ -2316,52 +2316,73 @@ public void setHCDetectorTaskDone(ADTask adTask, ADTaskState state, ActionListen
             adTaskCacheManager.removeHistoricalTaskCache(detectorId);
         });
 
+        long timeoutInMillis = 2000;// wait for 2 seconds to acquire updating HC detector task semaphore
```

Comment:
From my understanding this is related to how long it takes for all of the entities to finish. Is it sufficient for a large number of entities, or do you expect this to be enough time even in a worst-case scenario?

Reply:
Discussed offline, understood now this is a limit of how long it takes the last entity to finish updating the task, rather than a limit of the number of tasks. During testing, the avg time was < 40ms, so seems 2s is a sufficient timeout.